### ISSUE TITLE
Consumer options `external.internal.topics` breaks consumer.SubscribeTopics()

### ISSUE THREAD

**Issue by larrycinnabar** [confluentinc/confluent-kafka-go#186]
Description
===========
I'd like to read `__consumer_offsets` topic data. If I add `external.internal.topics` to consumer options (btw, no matter if it's true or false), the `consumer.SubscribeTopics()` fails

How to reproduce
================
```golang
	client, err := kafka.NewConsumer(&kafka.ConfigMap{
		"session.timeout.ms":              6000,
		"go.events.channel.enable":        true,
		"go.application.rebalance.enable": true,
		"exclude.internal.topics":         false, // btw, no matter false or true - will fail if it's presented anyway
		"default.topic.config":            kafka.ConfigMap{"auto.offset.reset": "earliest"},

		// other options on broker connections
	}, )
	if err != nil {
		return err
	}

        client.SubscribeTopics([]string{"__consumer_offsets"}, nil) //  fails
```

returns:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x40e386b]

goroutine 6 [running]:
.../vendor/github.com/confluentinc/confluent-kafka-go/kafka.(*Consumer).SubscribeTopics(0x0, 0xc420123120, 0x1, 0x1, 0x0, 0x0, 0x0)
        /.../vendor/github.com/confluentinc/confluent-kafka-go/kafka/consumer.go:80 +0x10b
```

when trying to get `c.handle.rk` -  `c.handle` is nil

P.S. Using Go 1.10.2 and latest confluent-kafka-go (connected to confluent cloud)

**Comment by rnpridgeon** [comment]
@larrycinnabar, I don't believe `exclude.internal.topics` is configuration property recognized by librdkafka or its derived clients. Only the old scala based consumers and 0.10+ Java consumers support the property as I understand it. 

With that said I would have expected an error to be raised when converting this property the c configuration which should have prevented the instantiation of the consumer. What happens when you leave this property out of the configuration map?

**Comment by larrycinnabar** [comment]
No error when the property is not present. So, this property is just not supported and assumed as "non-existed" - that's why handle is not inited no matter if prop is true or false
btw, maybe lib should return an error instead of panic ("property doesn't exist...")

**Comment by larrycinnabar** [comment]
Is there another way to get values from `__consumer_offsets`?

**Comment by rnpridgeon** [comment]
Once subscribed to the topic you should just be able to call ReadMessage in a loop. 

e.g.
```
	c.SubscribeTopics([]string{"__consumer_offsets"}, nil)

	for {
		msg, err := c.ReadMessage(100 * time.Millisecond)
		if err == nil {
			fmt.Printf("consumed: %s: %s\n", msg.TopicPartition, string(msg.Value))
		}
	}
```

If you are not seeing any messages this is perhaps because the default value for `auto.offset.reset` is latest. This means you will only see messages as they are appended from the the time you subscribed to this topic.  

You could optionally change this setting to `earliest` with the following config property: 

"default.topic.config": kafka.ConfigMap{"auto.offset.reset": "earliest"}

Note: You will need to write some sort of decoder to make sense of the __consumer_offsets topic's contents. Java clients for instance can use the following:

https://github.com/confluentinc/kafka/blob/trunk/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L1245-L1263

**Comment by larrycinnabar** [comment]
ok, both:
`"default.topic.config": kafka.ConfigMap{"auto.offset.reset": "earliest"}`
`auto.offset.reset":"earliest",` 
are set. 

No messages are coming
`ReadMessage` returns error code -185 without message body.

**Comment by rnpridgeon** [comment]
Error Code -185 means the request has timed out. 

https://github.com/confluentinc/confluent-kafka-go/blob/master/kafka/consumer.go#L316-L319

https://github.com/edenhill/librdkafka/blob/6c722e5d411882283e5687b3e214075804e782ae/src/rdkafka.h#L284-L285

Just to confirm are you able to consume from any topic or is this the first time you have tried consuming from the cluster with a go client?

**Comment by larrycinnabar** [comment]
> Just to confirm are you able to consume from any topic or is this the first time you have tried consuming from the cluster with a go client?

I'm consuming successfully from other topics (but via channel-based consumer).
Now I need a separate application that will just calculate and log consumers lag

**Comment by rnpridgeon** [comment]
Thanks for confirming, I'll try to reproduce this locally and get back to you.

**Comment by larrycinnabar** [comment]
couple more details:
1) I have an extra layer upon confluent-kafka-go. So, I've tried one more time, directly calling `confluent-kafka-go`. So it's not a point anymore
2) I connect to confluent cloud

Code that I use finally:

```golang
		c, err := kafka.NewConsumer(&kafka.ConfigMap{
			"bootstrap.servers":       *kafkaNodes,
			"group.id":                "my-group-name",
			"session.timeout.ms":      6000,
			"default.topic.config":    kafka.ConfigMap{"auto.offset.reset": "earliest"},
			"broker.version.fallback": "0.10.0.0",
			"api.version.fallback.ms": "0",
			"sasl.mechanisms":         "PLAIN",
			"security.protocol":       "SASL_SSL",
			"sasl.username":           *confluentUserName,
			"sasl.password":           *confluentPassword,
		})

		if err != nil {
			fmt.Fprintf(os.Stderr, "Failed to create consumer: %s\n", err)
			os.Exit(1)
		}

		fmt.Printf("Created Consumer %v\n", c)

		err = c.SubscribeTopics([]string{"__consumer_offsets"}, nil)
		log.Printf("%#v", err)

		for {
			msg, err := c.ReadMessage(100 * time.Millisecond)
			if err == nil {
				fmt.Printf("consumed: %s: %s\n", msg.TopicPartition, string(msg.Value))
			}
		}

		fmt.Printf("Closing consumer\n")
```

**Comment by larrycinnabar** [comment]
any news on reproducing the issue?
